File: CS238/js/teu.py
import math

def teu_loops(i):
    val = 1
    sum = 0
    ones = 0

    while val > 0:
        # print('started loop')
        if ones > 1:
            # print('ones greater than 1')
            return False
        if val % 10 == 1:
            ones += 1
            # print('ones:', ones)
        sum += val % 10
        # print(sum)
        val = math.floor(val / 10)
    if sum != 17:
        return False
    return True


counter = 0
for i in range(0, 1000000):
    if teu_loops(i):
        print('hello there')
        counter += 1

print( counter)

File: CS238/js/TEU.js
 // A function that counts solutions for a given predicate. 
 // Takes a max number of iterations, as well as a predicate. 
 // There is an optional third parameter, min, if you do not 
 // wish to start counting at zero
 function countSolutions(max, pred, min = 0) {
    let counter = 0;
    for (let i = min; i < max; i++) {
       pred(i) ? counter++ : null;
    }
    console.log("total",counter);
  }
 
 
  function teu_loops(i) {
    let val = i,
        sum = 0, 
        ones = 0;
   
    while (val) {
      if (ones > 1) {
        return false;
      }
      if (val % 10 === 1) {
        ones++;
      }
      sum += val % 10;
      val = Math.floor(val / 10);
    }
    if (sum != 17){
      return false;
    }
    return true;
  }
  countSolutions(1000000,teu_loops);